### ISSUE TITLE
maxwidth adding trailing comma to Dictionary type sugar syntax

### ISSUE THREAD

**Issue by ChristopherRogers** [nicklockwood/SwiftFormat#505]
Sorry, just found another formatting problem.

```
func foo(_ a: () -> [String: String]) {}
```

will be formatted as

```
func foo(
    _ a: () -> [
        String: String,
    ]
) {}
```

with `--maxwidth 15 --wraparguments beforefirst --wrapcollections beforefirst`.

**Comment by nicklockwood** [comment]
This seems to be a previously undiscovered bug in the trailingCommas rule. I'll fix it.

**Comment by nicklockwood** [comment]
@ChristopherRogers fixed in 0.41.2. Thanks 👍
